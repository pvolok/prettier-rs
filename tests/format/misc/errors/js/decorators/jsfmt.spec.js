import { outdent } from "outdent";

run_spec(
  {
    importMeta: import.meta,
    snippets: [
      outdent`
        @decorator1
        export @decorator2 class A{}
      `,
      outdent`
        @decorator1
        export default @decorator2 class A{}
      `,
      "@decorator1 export @decorator2 class A {}",
      "@decorator1 export default @decorator2 class A {}",
      "export @decorator2 default class A {}",
    ],
  },
  ["babel"],
);
