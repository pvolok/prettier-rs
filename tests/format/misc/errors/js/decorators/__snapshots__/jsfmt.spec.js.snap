// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 [babel] format 1`] = `
"Decorators can be placed *either* before or after the 'export' keyword, but not in both locations at the same time. (2:8)
  1 | @decorator1
> 2 | export @decorator2 class A{}
    |        ^"
`;

exports[`snippet: #1 [babel] format 1`] = `
"Decorators can be placed *either* before or after the 'export' keyword, but not in both locations at the same time. (2:16)
  1 | @decorator1
> 2 | export default @decorator2 class A{}
    |                ^"
`;

exports[`snippet: #2 [babel] format 1`] = `
"Decorators can be placed *either* before or after the 'export' keyword, but not in both locations at the same time. (1:20)
> 1 | @decorator1 export @decorator2 class A {}
    |                    ^"
`;

exports[`snippet: #3 [babel] format 1`] = `
"Decorators can be placed *either* before or after the 'export' keyword, but not in both locations at the same time. (1:28)
> 1 | @decorator1 export default @decorator2 class A {}
    |                            ^"
`;

exports[`snippet: #4 [babel] format 1`] = `
"Leading decorators must be attached to a class declaration. (1:20)
> 1 | export @decorator2 default class A {}
    |                    ^"
`;
